### PR TITLE
[receiver/azuremonitor]: Support Azure gov cloud

### PR DESCRIPTION
**Pull request by danelson**
**Description:** <Describe what has changed.>
Support Azure gov cloud

**Link to tracking Issue:** N/A

**Testing:** Ran against a government subscription

**Documentation:** Updated README

### CODE REVIEW

File: receiver/azuremonitorreceiver/config.go [hunk 1]
```diff
@@ -20,6 +20,10 @@ var (
 	errMissingClientID       = errors.New(`ClientID" is not specified in config`)
 	errMissingClientSecret   = errors.New(`ClientSecret" is not specified in config`)
 	errMissingFedTokenFile   = errors.New(`FederatedTokenFile is not specified in config`)
+	errInvalidCloud          = errors.New(`Cloud" is invalid`)
+
+	azureCloud           = "AzureCloud"
```
Comment (danelson) [inline receiver/azuremonitorreceiver/config.go:25]: I am not sure these are the preferred names, but these are the names returned from running `az cloud list`. Other tools use different conventions (e.g. terraform).

**Comment by danelson** [comment]
@codeboten @altuner can I get some feedback on this?

**Review by altuner — CHANGES_REQUESTED** [review]

File: receiver/azuremonitorreceiver/README.md [hunk 2]
```diff
@@ -17,37 +17,44 @@ This receiver scrapes Azure Monitor API for resources metrics.
...
 Using Service Principal for authentication:
+
 ```yaml
 receivers:
   azuremonitor:
     subscription_id: "${subscription_id}"
     tenant_id: "${tenant_id}"
     client_id: "${client_id}"
     client_secret: "${env:CLIENT_SECRET}"
+    cloud: AzureUSGovernment
```
[the remark this answers is not in the document]
Comment (danelson) [reply receiver/azuremonitorreceiver/README.md:57]: Not sure. I wouldn't expect users to include this line at all when configuring the public cloud. They can also infer this from the bullets above but I thought a more complete example was nice. Happy to use the default tho if you think that makes sense.

File: receiver/azuremonitorreceiver/factory.go [hunk 3]
```diff
@@ -18,6 +18,7 @@ import (
 
 const (
 	defaultCollectionInterval = 10 * time.Second
+	defaultCloud              = "AzureCloud"
```
[the remark this answers is not in the document]
Comment (danelson) [reply receiver/azuremonitorreceiver/factory.go]: Had to move config vars to constants. Let me know if that doesn't work

File: receiver/azuremonitorreceiver/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (danelson) [reply receiver/azuremonitorreceiver/README.md:57]: @altuner just following up here and with the other changes.

**Comment by danelson** [comment]
@altuner @codeboten sorry to keep pinging but I would love to move this along so that I can use it in my collector. Anything I can do to make that happen please let me know!

**Review by codeboten — APPROVED** [review]

File: receiver/azuremonitorreceiver/config.go [hunk 4]
```diff
@@ -225,6 +231,7 @@ var (
 type Config struct {
 	scraperhelper.ScraperControllerSettings `mapstructure:",squash"`
 	MetricsBuilderConfig                    metadata.MetricsBuilderConfig `mapstructure:",squash"`
+	Cloud                                   string                        `mapstructure:",cloud"`
```
Comment (codeboten) [inline receiver/azuremonitorreceiver/config.go:234]:
```suggestion
	Cloud                                   string                        `mapstructure:"cloud"`
```

**Review by codeboten — COMMENTED** [review]
See the suggestion

```
        	            	type "Config" from package "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/azuremonitorreceiver" has invalid config settings: field "Cloud" has config tag "" which doesn't satisfy "^[a-z0-9][a-z0-9_]*$"
```

**Pull request merged**